# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerPrivate.cpp
# Commit: e80b18ca9165
# Full Hash: e80b18ca916518c73d5435fe8f842d9f0e1b7585
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-25 21:02:33
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r cb743703a9ee -r e80b18ca9165 dom/workers/WorkerPrivate.cpp
--- a/dom/workers/WorkerPrivate.cpp	Tue Mar 25 09:11:22 2025 +0000
+++ b/dom/workers/WorkerPrivate.cpp	Tue Mar 25 09:13:20 2025 +0000
@@ -6463,7 +6463,7 @@
 }
 
 const nsAString& WorkerPrivate::Id() {
-  AssertIsOnMainThread();
+  AssertIsOnParentThread();
 
   if (mId.IsEmpty()) {
     mId = ComputeWorkerPrivateId();